File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
from collections import *
def solution(nums):
    answer = 0
    n = len(nums)//2
    cnt = Counter(num for num in nums)
    # 1:1 2:1, 4:2 n
    # 4 2
    answer = min(n, len(cnt))
    
    return answer

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    int len = nums.size();
    unordered_map <int, int> mp;
    for(auto n:nums){
        mp[n]++;
    }
    if(mp.size()<=len/2){
        return mp.size();
    }
    else{
        return len/2;
    }
    return answer;
}